packages/legend-application-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx

```diff
@@ -450,6 +431,131 @@ const ProjectDependencyEditor = observer(
   },
 );
 
+const PlatformBasicEditor = observer(
+  (props: {
+    platform: ProjectServerPlatform;
+    projectConfig: ProjectConfiguration;
+    isReadOnly: boolean;
+  }) => {
+    const { platform, projectConfig, isReadOnly } = props;
+
+    const changeVersion = action((val: string | undefined): void => {
+      projectConfig.changePlatformVersion(platform, val ? val : '');
+    });
+
+    const isNotValidPlatformVersion = action(
+      (val: string): boolean => val === '',
+    );
+
+    return (
+      <PanelFormTextEditorWithValidation
+        isReadOnly={isReadOnly}
+        value={platform.platformVersion}
+        name={platform.name}
+        update={changeVersion}
+        validationErrorMessage={
+          isNotValidPlatformVersion(platform.platformVersion)
+            ? 'Version cannot be empty'
+            : undefined
+        }
+      />
+    );
+  },
+);
+
+const ProjectPlatformVersionEditor = observer(
+  (props: { projectConfig: ProjectConfiguration; isReadOnly: boolean }) => {
+    const { projectConfig, isReadOnly } = props;
+    const editorStore = useEditorStore();
+
+    const platformConfigurations = projectConfig.platformConfigurations;
+
+    if (platformConfigurations) {
```

Comment:
This is genrally bad pattern of code, you're modifying the state in the body of the component. The concern here is not proper either. If you think about it, `overrideDefaultPlatform` is a synthetic boolean to describe whether `platformConfigurations === undefined`. Please clean this up. I don't think we need `overridesDefaultPlatform` at all

Reply:
true 🥲